pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -778,3 +778,11 @@ def test_slice_duplicate_monotonic(self):
         result = idx._maybe_cast_slice_bound('2017-01-01', 'left', 'loc')
         expected = Timestamp('2017-01-01')
         assert result == expected
+
+    def test_split_date_range_with_timezone(self):
+        # https://github.com/pandas-dev/pandas/issues/14042
+        idx = DatetimeIndex(['2016-01-01 00:00:00', '2016-01-02 00:00:00'],
+                            tz='Asia/Seoul')
+        split = np.split(idx, indices_or_sections=[])
```

Comment:
I found that Series does not suffer from this issue. But I'll leave the testcase.